### ISSUE TITLE
Release 2.18.0.0

### ISSUE THREAD

**Issue by rlueckl** [opensearch-project/opensearch-prometheus-exporter#317]
OpenSearch 2.18.0 has been released, see [release notes](https://github.com/opensearch-project/opensearch-build/blob/main/release-notes/opensearch-release-notes-2.18.0.md).
Official OpenSearch release ticket: https://github.com/opensearch-project/opensearch-build/issues/5004

New features:
- https://github.com/Aiven-Open/prometheus-exporter-plugin-for-opensearch/pull/322
- https://github.com/Aiven-Open/prometheus-exporter-plugin-for-opensearch/pull/323
- https://github.com/Aiven-Open/prometheus-exporter-plugin-for-opensearch/pull/246

Checklist:

- [x] [Compatibility Matrix](https://github.com/aiven/prometheus-exporter-plugin-for-opensearch#compatibility-matrix) is updated (TODO: link to PR, https://github.com/Aiven-Open/prometheus-exporter-plugin-for-opensearch/pull/326)
- [x] Release [2.18.0.0 tag](https://github.com/aiven/prometheus-exporter-plugin-for-opensearch/releases/tag/2.18.0.0) is present in the repo
- [x] Release artifacts are available and downloadable
- [x] [Release notes](https://github.com/aiven/prometheus-exporter-plugin-for-opensearch/releases/tag/2.18.0.0) published
- [ ] Forum announcement published

**Comment by rlueckl** [comment]
Hi! OpenSearch 2.18.0 was released on 2024-10-28, so I thought I'd open a ticket for a new prometheus-exporter version as well. I hope that's okay.

**Comment by ektropy** [comment]
There is a PR for this, #316.

**Comment by lukas-vlcek** [comment]
Hi, let me see if I can merge both https://github.com/Aiven-Open/prometheus-exporter-plugin-for-opensearch/pull/295 and https://github.com/Aiven-Open/prometheus-exporter-plugin-for-opensearch/pull/246 into master and let's do release asap.

**Comment by Biowoolf** [comment]
will be great

**Comment by Biowoolf** [comment]
> Hi, let me see if I can merge both #295 and #246 into master and let's do release asap.

how is it going?

**Comment by Biowoolf** [comment]
@lukas-vlcek , @vvanholl  how is it going? is there some blocks to merge and release ?

tell us please =)

**Comment by coredump17** [comment]
any updates on 2.18 release?

**Comment by zawistowskipiotr** [comment]
do we have any ETA on the release date? @lukas-vlcek

**Comment by benamazing** [comment]
same question, is there any update?

**Comment by Biowoolf** [comment]
@lukas-vlcek , @vvanholl UP
Is there any updates since November?

**Comment by patelsmit32123** [comment]
@xnox please merge #295

**Comment by igor-nikiforov** [comment]
@lukas-vlcek Just a friendly reminder

**Comment by suriyanath** [comment]
@lukas-vlcek  please release 2.18.0 waiting for this exporter to monitor my latest deployment

**Comment by xnox** [comment]
> @xnox please merge #295

Hi, I am not upstream developer of Aiven-Open. The commits and PRs you see referenced simply rebuild this plugin for 2.18.0.0 without any other changes for Wolfi OS & Chainguard Images.

**Comment by dcorneli** [comment]
Yes just pull the project and build it yourself, the only thing that changed afaik is the version number, I've tested and it works.

**Comment by AndreVirtimo** [comment]
We forked the repo and released a 2.18.0 version -> https://github.com/Virtimo/prometheus-exporter-plugin-for-opensearch/releases/tag/v2.18.0

**Comment by spapadop** [comment]
@lukas-vlcek thanks for merging https://github.com/Aiven-Open/prometheus-exporter-plugin-for-opensearch/pull/326 and adding BWC!I guess you still need to tag https://github.com/Aiven-Open/prometheus-exporter-plugin-for-opensearch/releases/tag/2.18.0.0 ?

**Comment by kingnarmer** [comment]
Any updates on when exporter for 2.18.0 will be published ?

**Comment by nicolechec** [comment]
@AndreVirtimo  please release 2.18.0 waiting for this exporter to grafana dashboards,thanks

**Comment by AndreVirtimo** [comment]
> @AndreVirtimo please release 2.18.0 waiting for this exporter to grafana dashboards,thanks

In our fork we have already released 2.18.0.
You can find it on https://github.com/Virtimo/prometheus-exporter-plugin-for-opensearch/releases/tag/v2.18.0

**Comment by StewartWBrown** [comment]
Hi @lukas-vlcek , apologies for yet another tag on this issue but as the Aiven-open prometheus-exporter plugin is the one that we have reliably been using for some time, we are wondering the status of this project moving forward? 

Are there plans to continue to release future versions of the Prometheus exporter under this repo? I see the upgrade exporter ticket has been merged recently, and an update to tests, yet no actual release yet. 

If this project looks to become inactive, we could continue discussion on this ticket (https://github.com/opensearch-project/OpenSearch/issues/8990#) as it seems quite a few folk are dependent on the release of a Prometheus exporter plugin. 

Regardless, appreciate the work you have put in to timely release this plugin for OpenSearch versions for almost three years! Clarity on the future of this project would be appreciated! :)

**Comment by lukas-vlcek** [comment]
@StewartWBrown thanks for pinging me. The release shouldn't take long now (it mostly means merging 2 remaining community PRs mentioned in the ticket description, currently I am in the middle of merging #295 and adding BWC test for it), I have a full capacity to work on it now. Please keep an eye on this ticket for updates.

As for the future of this plugin I plan to open a separate ticket for it and explain in more details. The short gist of the future plan is that I would like to see if plugin can be fully replaced by [Metrics Framework](https://opensearch.org/docs/latest/monitoring-your-cluster/metrics/getting-started/) (MF) which could possibly make this plugin fully obsolete (that would be an ideal outcome IMO). If there is anything missing in MF then depending on what it is I could try to contribute to OpenSearch in order to fill the gaps.

I know that community has been waiting for this release for too long... thanks for patience. In the meantime I was able to add BWC tests #322 that I consider important for rolling upgrades as new features are added to this plugin. Release 2.18.0.0 is actually very different "animal" than past maintenance releases, this is why it was not going as smooth as I was hoping for 🙏 .

HTH

**Comment by AndreVirtimo** [comment]
FYI we keep releasing the current state of the plugin under https://github.com/Virtimo/prometheus-exporter-plugin-for-opensearch/releases until this project is releasing new versions or we run into a breaking change.

**Comment by d-mankowski-synerise** [comment]
> Release 2.18.0.0 is actually very different "animal" than past maintenance releases, this is why it was not going as smooth as I was hoping for

It is not. It is totally arbitrary decision from your side @lukas-vlcek to incorporate #322 and #295 into this release. Exporter for 2.18.0.0, 2.19.0.0 can be easily built and released with just changing 1-2 lines of code, by simply bumping the number - as has been proven by the fork of this repository made by @AndreVirtimo. 

99% of users would rather have up-to-date OpenSearch - and working metrics exporter, on which, their entire monitoring is based - rather than have to wait God-knows-how-long to merge some changes with extra features (which are nice, but not as important as new features from latest OpenSearch releases).

Not that I want to sound demanding or ungrateful - creating and maintaining an open-source project is hard and takes a lot of time. However, this exporter has become de facto the standard in terms of monitoring OpenSearch and lots of users/companies using OpenSearch are relying on it.

And in this particular case - I find it simply unreasonable. There is literally no reason not to bump version of exporter, and THEN merge the PRs you mentioned, especially, since bumping exporter makes basically 0 changes to the code base.

**Comment by StewartWBrown** [comment]
Hey all- it's been 6 months since a regular exporter release, and everyone seems to be doing their own method of gathering Prometheus metrics for newer OpenSearch releases. I see that @spapadop has raised a technical-steering ticket to discuss future OpenSearch support for Prometheus metrics. 

If being unable to consume a new OpenSearch prometheus-exporter-plugin version has affected you at all, please show engagement /thumbs up/add thoughts to this issue:

https://github.com/opensearch-project/technical-steering/issues/35
